src/DocumentDB.ChangeFeedProcessor/FeedProcessing/ChangeFeedObserverFactory.cs

```diff
@@ -9,7 +9,7 @@ internal class ChangeFeedObserverFactory<T> : IChangeFeedObserverFactory
     {
         public IChangeFeedObserver CreateObserver()
         {
-            return new T();
+            return new ChangeFeedObserver(new T());
```

Comment:
not bad but wouldn't work when factory is provided by the user. Can we wrap observer with decorator at the time we call .createObserver? Or is the factory at that time is not when we need?